src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -1285,37 +1291,35 @@ private boolean contains(TableMetadata metadata, DecoratedKey partitionKey, Row
             else
             {
                 ByteBuffer foundValue = getValue(metadata, partitionKey, row);
-                if (foundValue == null)
-                    return false;
-
-                switch (type.kind)
-                {
-                    case LIST:
-                        ListType<?> listType = (ListType<?>)type;
-                        return listType.compose(foundValue).contains(listType.getElementsType().compose(value));
-                    case SET:
-                        SetType<?> setType = (SetType<?>)type;
-                        return setType.compose(foundValue).contains(setType.getElementsType().compose(value));
-                    case MAP:
-                        MapType<?,?> mapType = (MapType<?, ?>)type;
-                        return mapType.compose(foundValue).containsValue(mapType.getValuesType().compose(value));
-                }
-                throw new AssertionError();
+                return foundValue != null && Operator.CONTAINS.isSatisfiedBy(type, foundValue, value, indexAnalyzer, queryAnalyzer);
```

Comment:
That is so much better!

Reply:
Test coverage complains we cover 3 of 4 conditions only

Reply:
Is there a reason we cannot do a similar logical simplification in the `column.isComplex()` case?